### PR TITLE
[Handles] Updates to RPC

### LINKED ISSUE #1305: [Handles] require no handle for msa on retirement

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is  following
- [x] Add a new RPC `get_msa_for_handle` that retrieves MSA id from full `display_handle` e.g. Alice.11
- [x] Updated RPC `get_next_suffixes` to accept a string and optional number as count instead of payload
- [x] Update and add integration tests
- [x] Fix and update unit tests
- [x] Run all tests

Priority to merge in is low :yellow_circle: 

Closes #1305 #1364 #1369 #1371 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: common/primitives/src/handles.rs [hunk 1]
```diff
@@ -18,6 +18,8 @@ pub const HANDLE_BASE_CHARS_MAX: u32 = 20;
 pub const HANDLE_BASE_BYTES_MAX: u32 = 32; // Hard limit of 32 bytes
 /// The maximum number of digits in a suffix
 pub const SUFFIX_MAX_DIGITS: usize = 5; // The max value of a HandleSuffix (u16) is 65535 which is 5 digits.
+/// The maximum count of suffixes allowed to be requested at once
+pub const MAX_SUFFIXES_COUNT: u16 = 100;
```
Comment (saraswatpuneet) [inline common/primitives/src/handles.rs:22]: Keeping max allowed suffixes to be requested to 100

File: pallets/handles/src/rpc/src/lib.rs [hunk 2]
```diff
@@ -80,11 +85,23 @@ where
...
-		handle_input: PresumptiveSuffixesRequest,
+		base_handle: String,
+		count: Option<u16>,
 	) -> RpcResult<PresumptiveSuffixesResponse> {
 		let api = self.client.runtime_api();
 		let at = BlockId::hash(self.client.info().best_hash);
-		let suffixes_result = api.get_next_suffixes(&at, handle_input);
+		let handle_input: Handle = base_handle.into_bytes().try_into().unwrap();
+		let max_count = MAX_SUFFIXES_COUNT;
+		let count = count.unwrap_or(max_count).min(max_count);
```
Comment (saraswatpuneet) [inline pallets/handles/src/rpc/src/lib.rs:95]: Take lowest of two

File: pallets/handles/src/rpc/src/tests/mod.rs [hunk 3]
```diff
@@ -23,13 +23,17 @@ sp_api::mock_impl_runtime_apis! {
...
 			let mut suffixes = Vec::new();
 			for i in 0..suffix_request.count {
 				suffixes.push(i);
 			}
 			PresumptiveSuffixesResponse { base_handle: suffix_request.base_handle,  suffixes }
 		}
+
+		fn get_msa_for_handle(_display_handle: Handle) -> Option<MessageSourceId>{
+			Some(VALID_MSA_ID)
+		}
```
Comment (saraswatpuneet) [inline pallets/handles/src/rpc/src/tests/mod.rs:36]: not so important unit test, good for coverage :)

File: js/api-augment/definitions/handles.ts [hunk 4]
```diff
@@ -10,12 +10,26 @@ export default {
       ],
       type: "Option<HandleResponse>",
     },
+    getMsaForHandle: {
+      description: "Get msa_id for a given handle",
+      params: [
+        {
+          name: "display_handle",
```
Comment (wilwade) [inline js/api-augment/definitions/handles.ts:17]: Isn't this a "full handle"? I'm still getting confused by these handle terms.

**Review by wilwade — COMMENTED** [review]
Few small things, but looking good.

File: js/api-augment/definitions/handles.ts (hunk 4 above)
Comment (saraswatpuneet) [reply js/api-augment/definitions/handles.ts:17]: Yes this is full_handle, we will cleanup the term `full_handle` instead keep `display_handle` across the code before final merge

File: pallets/handles/src/rpc/src/lib.rs (hunk 2 above)
Comment (wilwade) [reply pallets/handles/src/rpc/src/lib.rs:95]: Hmm... Let's not give them more than they asked for. `0` doesn't make sense, but if our default is `1` even if we increase it in the future we don't have a breaking change.

File: js/api-augment/definitions/handles.ts (hunk 4 above)
Comment (saraswatpuneet) [reply js/api-augment/definitions/handles.ts:17]: Added to Misc Todo on main PR and Robert has fixed most of it

File: pallets/handles/src/rpc/src/lib.rs [hunk 5]
```diff
@@ -80,11 +88,23 @@ where
...
-		handle_input: PresumptiveSuffixesRequest,
+		base_handle: String,
+		count: Option<u16>,
 	) -> RpcResult<PresumptiveSuffixesResponse> {
 		let api = self.client.runtime_api();
 		let at = BlockId::hash(self.client.info().best_hash);
-		let suffixes_result = api.get_next_suffixes(&at, handle_input);
+		let handle_input: Handle = base_handle.into_bytes().try_into().unwrap();
+		let max_count = MAX_SUFFIXES_COUNT;
+		let count = count.unwrap_or(DEFAULT_SUFFIX_COUNT).min(max_count);
```
Comment (saraswatpuneet) [inline pallets/handles/src/rpc/src/lib.rs:98]: `DEFAULT_SUFFIX_COUNT` is 1

**Pull request merged**